logic/pipeline.cpp

```diff
@@ -390,9 +390,20 @@ Pose pipeline::maybe_apply_filter(const Pose& value) const
 
 Pose pipeline::apply_zero_pos(Pose value) const
 {
-    for (int i = 0; i < 6; i++)
-        value(i) += m(i).opts.zero * (m(i).opts.invert ? -1 : 1);
 
+    const auto q = QQuaternion::fromEulerAngles(value[Pitch], value[Yaw], -value[Roll]);
+    const auto center = QQuaternion::fromEulerAngles(
+        m(Pitch).opts.zero * (m(Pitch).opts.invert ? -1 : 1),
+        m(Yaw).opts.zero * (m(Yaw).opts.invert ? -1 : 1),
+        m(Roll).opts.zero * (m(Roll).opts.invert ? 1 : -1)
+    );
+    const auto v = (center.conjugated() * q).toEulerAngles();
```

Comment:
It's good in principle, just add a separate setting for it and leave `apply_zero_pos` as it was.